serverless/serverless
# Config credentials command

## What did you implement:

Added a minimal `sls init` command to create the `default` profile on the user's file system without the AWS CLI, or any profile that they want.

Takes a `--provider` options which only accepts `aws` at the moment.  It also takes `--key` and `--secret` options.   Also takes an optional `--profile/-n` option. 

```
sls init -p aws -k JAJHLJFIAFJADOJO -s Jj298jafj9p // created "default" profile
sls init -p aws -k JAJHLJFIAFJADOJO -s Jj298jafj9p -n dev // created "dev" profile
```

Closes #2623

This plugin does safe-checking and does not overwrite anything. 
## How did you implement it:

Created a new init plugin in core.
## How can we verify it:
- Copy the contents of `~/.aws/credentials` to a temporary location
- Delete your `~/.aws/credentials` file
- Run this: `sls init -p aws -k JAJHLJFIAFJADOJO -s Jj298jafj9p`
- Check `~/.aws/credentials` to see if a default profile was created.
- Run this: `sls init -p aws -k JAJHLJFIAFJADOJO -s Jj298jafj9p -n test1`
- Check `~/.aws/credentials` to see if a test1 profile was created.

**Note:** Tests for creating the actual file are not currently implemented since this command writes to the user's root file, which I'm not sure Travis CI supports.  Further, this command will alter the profiles of the user running the tests locally, which presents additional problems.
## Todos:

**Ready for review:** YES
